scopes/dependencies/dependencies/dependencies-loader/auto-detect-deps.ts

```diff
@@ -578,34 +590,28 @@ export class AutoDetectDeps {
     this.debugDependenciesData.unidentifiedPackages = unidentifiedPackages;
   }
 
-  private _pushToDependenciesIfNotExist(
-    dependency: Dependency,
-    fileType: FileType,
-    depDebug: DebugComponentsDependency
-  ) {
+  private _pushToDependenciesIfNotExist(dependency: Dependency, opts: PushToDepsArrayOpts) {
     const existingDependency = this.getExistingDependency(this.allDependencies.dependencies, dependency.id);
     const existingDevDependency = this.getExistingDependency(this.allDependencies.devDependencies, dependency.id);
     // no need to enter dev dependency to devDependencies if it exists already in dependencies
-    if (existingDependency || (existingDevDependency && fileType.isTestFile)) {
+    if (existingDependency || (existingDevDependency && opts.fileType.isTestFile)) {
```

Comment:
I wonder what happens if you have a dependency without the always peer.
then you tag both components.
then you change the always peer to true.
maybe it will get into this condition and won't move to be peer?
worth validating

Reply:
added a test